### ISSUE TITLE
Fix missing request body

### ISSUE THREAD

**Issue by Janglinator** [kasketis/netfox#96]
As mentioned in the help docs, users have to manually write:

```
if let bodyData = mutableURLRequest.HTTPBody {
    NSURLProtocol.setProperty(bodyData, forKey: "NFXBodyData", inRequest: mutableURLRequest)
}
```

It would be spectacular if there was a way to do this internally using swizzling

**Comment by a-promise** [comment]
where the correct place wirte this?

Dose it means in the network framework?   then can get the body in the  func startLoading()? and remove it.

**Comment by Janglinator** [comment]
Honestly, I haven’t the slightest idea yet, this issue was more of a mental note until I find time to do so!

**Comment by cloudhm** [comment]
`(lldb) po self
▿ http://test.tw.whatsmode.com/platform-4.3-user/orders/info
  ▿ url : Optional<URL>
    ▿ some : http://test.tw.whatsmode.com/platform-4.3-user/orders/info
  - cachePolicy : 0
  - timeoutInterval : 60.0
  - mainDocumentURL : nil
  - networkServiceType : __ObjC.NSURLRequest.NetworkServiceType
  - allowsCellularAccess : true
  ▿ httpMethod : Optional<String>
    - some : "POST"
  ▿ allHTTPHeaderFields : Optional<Dictionary<String, String>>
    ▿ some : 5 elements
      ▿ 0 : 2 elements
        - key : "Content-Type"
        - value : "application/json"
      ▿ 1 : 2 elements
        - key : "Accept-Language"
        - value : "en;q=1, zh-Hant-US;q=0.9"
      ▿ 2 : 2 elements
        - key : "Content-Length"
        - value : "86"
      ▿ 3 : 2 elements
        - key : "Authorization"
        - value : "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiI2YWQ4NGZhNi05YjA5LTRlMzUtOWFiYS1hZDVkNzIwZTA1YzMiLCJjcmVhdGVkIjoxNTA4ODE0MzAyMDEyLCJleHAiOjE1MDkxNzQzMDJ9.eR5dNENgPn91Pju8ceZ2M5SLimK5e8zoO9qslEN-fJpdFHXtBQuNBhcdo8tuoVWxuJyYu1JW--n2H4P_rPLGRQ"
      ▿ 4 : 2 elements
        - key : "User-Agent"
        - value : "whatsmode/1.2.1 (iPhone; iOS 11.0; Scale/2.00)"
  - httpBody : nil
  ▿ httpBodyStream : Optional<NSInputStream>
  - httpShouldHandleCookies : true
  - httpShouldUsePipelining : false`

**Comment by a-promise** [comment]
Hello, I thought a solution to fix this bug。
URLRequest has a attribute named "hashcode" which can promise this request is unique.
so : 
```
override open class func canInit(with task: URLSessionTask) -> Bool {
        guard let request = task.originalRequest else { return false }
		requests[request.hashValue] = request
        return canServeRequest(request)
    }
```

```
override open func startLoading() {
             var req: NSMutableURLRequest
		req = ((MonitorURLProtocol.requests[self.request.hashValue] ?? self.request) as NSURLRequest).mutableCopy() as! NSMutableURLRequest
            // self.request  can be replaced by function self.canonical(forRequest:)
              ...
    }
```

this is a bug , if you fire many same request , there is only one request can be catched. 

if you want to get all requests. you can process this hashcode with the  request's emit time. Er..... @

**Comment by a-promise** [comment]
sorry, I lost remove request from requests  when the task is end.
 requests is a  static instant variable( Dictionary  )

**Comment by rajanbalana** [comment]
Fixed the issue in #102

**Comment by Abeansits** [comment]
Has the fix proposed by @rajanbalana been confirmed? His PR has been stuck for a while. @Janglinator
